evalml/automl/automl_search.py

```diff
@@ -205,22 +205,21 @@ def __init__(self,
         additional_objectives = [self._validate_objective(obj) for obj in additional_objectives]
         self.additional_objectives = additional_objectives
 
-        if max_time is None or isinstance(max_time, (int, float)):
-            self.max_time = max_time
-        elif isinstance(max_time, str):
-            self.max_time = convert_to_seconds(max_time)
-        else:
-            raise TypeError("max_time must be a float, int, or string. Received a {}.".format(type(max_time)))
-
+        if not isinstance(max_time, (int, float, str, type(None))):
+            raise TypeError(f"Parameter max_time must be a float, int, string or None. Received {str(max_time)}.")
+        if isinstance(max_time, (int, float)) and max_time <= 0:
+            raise ValueError(f"Parameter max_time must be None or non-negative. Received {max_time}.")
```

Comment:
Hmm, the `ValueError` message doesn't align with the check (max_time <= 0 vs non-negative)

Reply:
Maybe 'strictly positive' rather than 'non-negative'?

Reply:
Good point. I'll change the boundary conditions here to be non-negative and if we wanna change this in the future we can!